src/lightning_wallet.ts

```diff
@@ -0,0 +1,180 @@
+import {
+  AddressType,
+  GetInfoResponse,
+  Invoice,
+  OpenStatusUpdate,
+  SendResponse
+} from "@radar/lnrpc";
+import pEvent from "p-event";
+import { Lnd } from "./lnd";
+
+export class LightningWallet {
+  public static async newInstance(
+    certPath: string,
+    macaroonPath: string,
+    lndHost: string,
+    lndPort: number
+  ): Promise<LightningWallet> {
+    const lnd = await Lnd.init({
+      tls: certPath,
+      macaroonPath,
+      server: `${lndHost}:${lndPort}`
+    });
+    return new LightningWallet(lnd, lndHost, lndPort);
+  }
+
+  public constructor(
+    public readonly lnd: Lnd,
+    private readonly p2pHost: string,
+    private readonly p2pPort: number
+  ) {}
+
+  public async sendPayment(
+    publicKey: string,
+    satAmount: string,
+    secretHash: string,
+    finalCltvDelta: number
+  ): Promise<SendResponse> {
+    const publicKeyBuf = Buffer.from(publicKey, "hex");
+    const paymentHash = Buffer.from(secretHash, "hex");
+
+    return this.lnd.lnrpc.sendPaymentSync({
+      dest: publicKeyBuf,
+      amt: satAmount,
+      paymentHash,
+      cltvLimit: finalCltvDelta
+    });
+  }
+
+  public async addHoldInvoice(
+    satAmount: string,
+    secretHash: string,
+    expiry: number,
+    memo: string
+  ): Promise<string> {
+    const satAmountNum = parseInt(satAmount, 10);
+
+    return (
+      await this.lnd.invoicesrpc.addHoldInvoice({
+        value: satAmountNum,
+        hash: secretHash,
+        expiry,
+        memo
+      })
+    ).paymentRequest;
+  }
+
+  public async settleInvoice(secret: string): Promise<void> {
+    await this.lnd.invoicesrpc.settleInvoice({ preimage: secret });
+  }
+
+  public async newAddress(type: AddressType): Promise<string> {
+    return (await this.lnd.lnrpc.newAddress({ type })).address;
+  }
+
+  public async confirmedChannelBalance(): Promise<string> {
+    return (await this.lnd.lnrpc.channelBalance()).balance;
+  }
+
+  public async confirmedWalletBalance(): Promise<string> {
+    return (await this.lnd.lnrpc.walletBalance()).confirmedBalance;
+  }
+
+  public async getPubkey(): Promise<string> {
+    return (await this.lnd.lnrpc.getInfo()).identityPubkey;
+  }
+
+  public getLndP2pSocket(): string {
+    return `${this.p2pHost}:${this.p2pPort}`;
+  }
+
+  public async getInfo(): Promise<GetInfoResponse> {
+    return this.lnd.lnrpc.getInfo();
+  }
+
+  public async openChannel(
+    pubkey: string,
+    satAmount: number
+  ): Promise<Outpoint> {
+    const request = {
+      nodePubkey: Buffer.from(pubkey, "hex"),
+      localFundingAmount: satAmount.toString()
+    };
+    const openChannel = this.lnd.lnrpc.openChannel(request);
+
+    openChannel.on("error", (err: any) => {
```

Comment:
:mega: This should be a failure of the function.

Reply:
I do not understand the action you want me to take from this comment.

Reply:
I would like to merge without testing that because my mastery of the jest framework for this specific test is just not good enough.

I can make the error happen, I just cannot assert it.

`lightning_wallet.spec.ts` content:
```
import { EventEmitter } from "events";
import { LightningWallet } from "./lightning_wallet";

jest.mock("./lnd");

describe("Lightning Wallet", () => {
  it("Propagate exception from event handler", async () => {

    const emitter = new EventEmitter();

    const mockOpenChannel = jest.fn().mockImplementation(() => {
      return emitter;
    });

    const lightningWallet = await LightningWallet.newInstance("", "","","");

    // @ts-ignore
    lightningWallet.lnd = {
      lnrpc: {openChannel: mockOpenChannel}
    };

    const openChannelPromise = lightningWallet.openChannel("", 1000);

    const expectPromise = expect(openChannelPromise).rejects.toMatch("Error encountered for Open Channel");
    emitter.emit("error");
    await expectPromise;
  });
});
```

Running the test results to:
```
Error: Error encountered for Open Channel: undefined

    at EventEmitter.openChannel.on.err (/Users/froyer/src/comit-js-sdk/src/lightning_wallet.ts:2685:15)
    at EventEmitter.emit (events.js:203:15)
    at /Users/froyer/src/comit-js-sdk/src/lightning_wallet.spec.ts:31:13
    at Generator.next (<anonymous>)
    at fulfilled (/Users/froyer/src/comit-js-sdk/src/lightning_wallet.spec.ts:5:58)
    at process._tickCallback (internal/process/next_tick.js:68:7)
```

Which is as expected.

Moreover, this test is not really about testing our code but it is about confirming that errors thrown in the callback of event listener are propagated. My patch above does prove it so I suggest we just move forward :)

Reply:
> Moreover, this test is not really about testing our code but it is about confirming that errors thrown in the callback of event listener are propagated. My patch above does prove it so I suggest we just move forward :)

Agreed, let's move forward!